src/TemporalExpression/WeekInYear.php

```diff
@@ -0,0 +1,38 @@
+<?php
+namespace Riskio\Schedule\TemporalExpression;
+
+use DateTime;
+
+class WeekInYear implements TemporalExpressionInterface
+{
+    /**
+     * @var int
+     */
+    protected $weekIndex;
+
+    /**
+     * @param int $weekIndex
+     */
+    public function __construct($weekIndex)
+    {
+        $filtered = filter_var($weekIndex, FILTER_VALIDATE_INT, [
+            'options' => ['min_range' => 1, 'max_range' => 53],
+        ]);
+        if (!$filtered) {
+            throw new Exception\InvalidArgumentException(
```

Comment:
`InvalidWeekNumberException` would be more meaningful and would allow the caller to construct multiple temporal expression at once and be able to catch specific error